lib/sidekiq/worker.rb

```diff
@@ -351,12 +347,31 @@ def sidekiq_options(opts = {})
         super
       end
 
-      def client_push(item) # :nodoc:
+      def client_push(stringified_item) # :nodoc:
         pool = Thread.current[:sidekiq_via_pool] || get_sidekiq_options["pool"] || Sidekiq.redis_pool
-        stringified_item = item.transform_keys(&:to_s)
-
         Sidekiq::Client.new(pool).push(stringified_item)
       end
+
+      def serialize(opts = {}, args = [])
+        new(opts, *args).serialize
+      end
```

Comment:
This is an interface, but would you say that it's a public interface that Sidekiq extensions can rely on?

1. it's a method inside the module `ClassMethods`
2. `ClassMethods` gets included in each worker where `serialize` becomes a class method
3. The topic of public / private class methods is "interesting" in Ruby and you need to go through some hoops to make class methods private (main reason why I don't like them).
4. Therefore this is not [boring Ruby code](https://blog.sundaycoding.com/blog/2018/11/22/boring-ruby-code/) which I like.
5. IMO we can talk about a public interface when there's a public instance method, preferably in a separate class.

I'm not sure if creating such a public interface was your intention or not, but I thought that I could leave a few thoughts.

Reply:
This exists because essentially all of the Sidekiq worker interface exists as class methods. Thus, to make use of this new feature from `Job.perform_async`, I needed a class method. Since I needed it, I figured it likely useful to expose (in the same way that `#serialize` is publicly exposed).